api_tests/lib_sdk/wallets/index.ts

```diff
@@ -42,16 +50,36 @@ export class Wallets {
         return wallet;
     }
 
-    public async initializeForLedger<K extends keyof AllWallets>(name: K) {
+    public async initializeForLedger<K extends keyof AllWallets>(
+        name: K,
+        logger: Logger,
+        logDir: string,
+        actorConfig: E2ETestActorConfig
+    ) {
         switch (name) {
             case "ethereum":
                 this.wallets.ethereum = new EthereumWallet(
                     global.ledgerConfigs.ethereum
                 );
                 break;
             case "bitcoin":
-                this.wallets.bitcoin = await BitcoinWallet.newInstance(
-                    global.ledgerConfigs.bitcoin
+                if (!this.wallets.bitcoin) {
+                    this.wallets.bitcoin = await BitcoinWallet.newInstance(
+                        global.ledgerConfigs.bitcoin
+                    );
+                }
+                break;
+            case "lightning":
+                if (!this.wallets.bitcoin) {
+                    this.wallets.bitcoin = await BitcoinWallet.newInstance(
+                        global.ledgerConfigs.bitcoin
+                    );
+                }
+                this.wallets.lightning = await LightningWallet.newInstance(
+                    this.wallets.bitcoin,
```

Comment:
The LN wallet could just have its own wallet, no need to reuse them.

Reply:
I don't see the point in instantiated 2 different bitcoin wallets.

Reply:
Avoiding conditionals doesn't convince you? :(